### ISSUE TITLE
jetty-12 ee10 ServletApiResponse.sendError does not check for the response already being committed

### ISSUE THREAD

**Issue by janbartel** [eclipse/jetty.project#9648]
The javadoc for `HttpServletResponse.sendError` says:

```
  * If the response has already been committed, this method throws an IllegalStateException. After using this method, the
  * response should be considered to be committed and should not be written to.
```

Jetty is not checking if the response has already been committed and thus never throws the `IllegalStateException`.

Tck tests affected:

> com/sun/ts/tests/servlet/api/jakarta_servlet_http/httpservletresponse/URLClient.java.sendErrorIllegalStateExceptionTest
> com/sun/ts/tests/servlet/api/jakarta_servlet_http/httpservletresponse/URLClient.java.sendError_StringIllegalStateExceptionTest
> com/sun/ts/tests/servlet/pluggability/api/jakarta_servlet_http/httpservletresponse/URLClient.java.sendErrorIllegalStateExceptionTest
> com/sun/ts/tests/servlet/pluggability/api/jakarta_servlet_http/httpservletresponse/URLClient.java.sendError_StringIllegalStateExceptionTest

**Comment by olamy** [comment]
few weeks ago I have started a branch here https://github.com/eclipse/jetty.project/tree/jetty-12.0.x-should-throw-ise-after-flush
but didn’t really finish it.

**Comment by janbartel** [comment]
Fixed via #9669
